### PR TITLE
Fixes #1468: Port error page from Gatsby to Nextjs

### LINKED ISSUE #1468: Port Error page from GatsbyJS to Next.js
**What would you like to be added**:
This is an issue to port Error page from Gatsby to NextJS

**Why would you like this to be added**:
Part of #1316, transitioning of the Telescope front-end from Gatsby to Next

### PR DESCRIPTION
**Pull request by rjayroso**
## Issue This PR Addresses
Fixes #1468 
Closes #1506 (Old PR for same issue)

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
This PR takes over and continues the work of #1506 to port the error page from Gatsby to Next.js. 

### HOW TO TEST:
Set your URL to `/error`. 
**For example,** if I am running Telescope on `localhost:8000`, then my full URL will look like `http://localhost:8000/error`.  

To see default status code messages (400-405), add `?status=CODE` to `http://localhost:8000/error` where `CODE` is a number from 400-405. 
**For example:**  `http://localhost:8000/error?status=404` will result in displaying status code 404 and message 'We could not find what you were looking for!'
Note, codes out of this range will still be displayed and also it will default to message 'Something went wrong!'. 

To see custom messages, add `&message=MESSAGE` to `http://localhost:8000/error?status=CODE` where `CODE` is any number, and `MESSAGE` is a string.
**For example:** `http://localhost:8000/error?status=999&message=Server blew up!`

### NOTE
`PageBase` component was removed, the `_app.tsx` replaces its purpose.  

Next.js on the left, Gatsby on the right.
image
As you can see, without the `PageBase` component everything is still there (*minus the DynamicImage, I removed it for now as it caused mobile styling issues*)

## Checklist
- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

## Screenshots
Test with status 400
image
---
Test with no search parameters
image
---
Test with status 405 and custom message
image
---
Test with status 404 on mobile
image

### CODE REVIEW

**Comment by HyperTHD** [comment]
@rjayroso Use the FeedHash interface in `interfaces.ts` as a reference for how you can fix the typing for Proxy. You're going to need to do that so since the proxy object looks like it needs to be indexed. We'll hopefully get an update on PageBase in today's meeting so that you won't have that linting issue since it doesn't exist yet

**Comment by rjayroso** [comment]
> @rjayroso Use the FeedHash interface in `interfaces.ts` as a reference for how you can fix the typing for Proxy. You're going to need to do that so since the proxy object looks like it needs to be indexed. We'll hopefully get an update on PageBase in today's meeting so that you won't have that linting issue since it doesn't exist yet

Thank you so much, I'll take a look at that right now and see. I'll be working on the layout later today

File: src/frontend/next/src/pages/error.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    },
+  },
+}));
+
+type ErrorProps = {
+  message: string | null;
+  status: number | null;
+};
+
+function CreateInnerErrorContent({ message, status }: ErrorProps) {
```
Comment (humphd) [inline src/frontend/next/src/pages/error.tsx:101]: Drop the `Create-` prefix: `InnerErrorContent` is probably good.

File: src/frontend/next/src/pages/error.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+      {message}
+    </Typography>
+  );
+}
+
+const ErrorPage = () => {
+  const classes = useStyles();
+  const router = useRouter();
+
+  const baseURL = 'https://telescope.cdot.systems/';
```
Comment (humphd) [inline src/frontend/next/src/pages/error.tsx:145]: We need to get this from `config.ts` so we can configure it for local vs. staging vs. prod.

**Review by humphd — CHANGES_REQUESTED** [review]

**Review by rjayroso — COMMENTED** [review]
Dropped "Create" from `CreateInnerErrorContent` to `InnerErrorContent`, and added an import from `config.ts` to take the URL from there

File: src/frontend/next/src/pages/error.tsx [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+import { Card, CardActions, CardContent, Fab, Grid, Typography } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import ArrowBack from '@material-ui/icons/ArrowBack';
+import config from '../config';
+
+const useStyles = makeStyles((theme) => ({
+  container: {
+    position: 'absolute',
+    [theme.breakpoints.between('xs', 'sm')]: {
+      top: '0vh',
```
Comment (humphd) [inline src/frontend/next/src/pages/error.tsx:11]: `top: 0` (no units on 0, since 0px == 0vh == 0% etc)

**Review by humphd — CHANGES_REQUESTED** [review]
Let's get this rebased so vercel can build and deploy it for testing.

**Review by humphd — APPROVED** [review]

**Review by chrispinkney — APPROVED** [review]
Awesome job man!

**Review by UsernameHasBeenChanged100Times — APPROVED** [review]

**Pull request merged**